avoid OpOverloadPacket.__getattr__ calls in inductor lowering

bdhirsh opened the pull request:
we have seen stacktrace samples showing that a lot of compilation time is spent in exceptions raised in `OpOverloadPacket.__getattr__`. It's not entirely clear why/how this happens, but I spot-checked a few places in `_inductor.graph.py` where we previously may have been calling `hasattr(OpOverloadPacket, ...)`, that can be avoided (hasattr will go through getattr, which, for OpOverloadPacket, will do a lookup in the dispatch table for all overload names of the packet).

Test Plan: CI

Differential Revision: D60048270


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

davidberard98 reviewed (approved):
Thanks! Fingers crossed this resolves the issues we're seeing

bdhirsh committed:
avoid OpOverloadPacket.__getattr__ calls in inductor lowering (#131348)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/131348

Test Plan: CI

Reviewed By: davidberard98

Differential Revision: D60048270

ezyang commented:
Are you going to fix the hasattr calls in torch/_functorch/partitioners.py as well?
